lntest/harness.go

```diff
@@ -1197,11 +1197,17 @@ func (n *NetworkHarness) CloseChannel(ctx context.Context,
 	// Wait until either the deadline for the context expires, an error
 	// occurs, or the channel close update is received.
 	select {
+	case err := <-errChan:
+		return nil, nil, err
 	case <-ctx.Done():
+		// Check if there's a more specific error available.
+		select {
+		case err := <-errChan:
+			return nil, nil, err
+		case <-time.After(time.Second):
+		}
```

Comment:
Can you explain why use this, I really didn't understand

Reply:
Canceling the ctx will cause one of the statements in the goroutine to error out (and therefore send a value in `errChan`) but it's undefined whether the receive from `errChan` or the receive from `ctx.Done()` will happen first. Therefore, if `ctx.Done()` is signalled we first check if there is a more specific error we can supply so that debugging test failures is easier.

On a second thought though, we can probably just remove the receive from `ctx.Done` since a canceled context will cause one of the method calls in the goroutine to fail anyway.